### PR TITLE
Refactored Login/Signin up Flow

### PR DESCRIPTION
**Pull request by colejoh**

### CODE REVIEW

File: src/actions/users.js [hunk 1]
```diff
@@ -2,139 +2,142 @@ import cookie from 'react-cookie';
...
-    };
+  return dispatch => {
+    return apiFetch(`users/auth/github/${code}`, {
+      method: 'POST'
+    })
+      .then(response => response.json())
+      .then(json => {
+        if (json.success) {
+          let { token, action, username } = json.data;
+          console.log(json);
```
Comment (kedarv) [inline src/actions/users.js]: can we remove console.logs as we find them? or at least put them behind a debug flag

File: src/actions/users.js [hunk 2]
```diff
@@ -2,139 +2,142 @@ import cookie from 'react-cookie';
...
+          if (token) {
+            console.log('got jwt via gh', token);
```
Comment (kedarv) [inline src/actions/users.js]: remove comment

File: src/actions/users.js [hunk 3]
```diff
@@ -2,139 +2,142 @@ import cookie from 'react-cookie';
...
+              messagePrefix = `created an account using`;
+            }
+            toastr.success(
+              'Success!',
+              `${messagePrefix} github account ${username}`
+            );
+            dispatch(loginFromJWT(token));
+          }
+        } else {
+          console.log('authUserWithGithub error:', json.message);
```
Comment (kedarv) [inline src/actions/users.js]: remove comment

File: src/components/Nav/Nav.js [hunk 4]
```diff
@@ -81,6 +37,10 @@ class Nav extends Component {
             <NavLink exact to="/faq" className="p-nav__nav_link">
               FAQ
             </NavLink>
+            {/* Had this in for testing */}
```
Comment (kedarv) [inline src/components/Nav/Nav.js:40]: What's this for?

File: src/pages/Login/LoginForm.js [hunk 5]
```diff
@@ -1,35 +1,67 @@
...
+          disabled={submitting}
+          full
+          onClick={() => onSubmit(this.state.email, this.state.password)}
+        >
+          Login
+        </Button>
+        <GithubLoginButton actionText="Login" />
+
+        <div className="flex margint h-center">
+          <Link to="/register">Register</Link> /{' '}
```
Comment (kedarv) [inline src/pages/Login/LoginForm.js:58]: What's `{' '}` for? Formatting? Ignore if intentional.

File: src/pages/Login/index.js [hunk 6]
```diff
@@ -1,74 +1,76 @@
...
+    return apiFetch('users/login', {
+      method: 'POST',
+      body: d
+    })
+      .then(response => response.json())
+      .then(json => {
+        if (json.success === false) {
+          throw new SubmissionError({ _error: json.message });
+        } else {
+          console.log(json.data.token);
```
Comment (kedarv) [inline src/pages/Login/index.js:28]: remove comment

File: src/pages/PasswordReset/index.js [hunk 7]
```diff
@@ -1,8 +1,8 @@
 import React, { Component } from 'react';
 import { SubmissionError } from 'redux-form'
 import { Redirect } from 'react-router-dom'
-import RequestPasswordResetForm from './RequestPasswordResetForm';
-import PerformPasswordResetForm from './PerformPasswordResetForm';
+// import RequestPasswordResetForm from './RequestPasswordResetForm';
+// import PerformPasswordResetForm from './PerformPasswordResetForm';
```
Comment (kedarv) [inline src/pages/PasswordReset/index.js]: do we need these imports still?

File: src/pages/PasswordReset/index.js [hunk 8]
```diff
@@ -56,10 +56,10 @@ class PasswordReset extends Component {
         let isTokenShowing = this.props.match.params.reset_token !== undefined;
         let requestReset =(<div>
             <p>Please enter your email.</p>
-            <RequestPasswordResetForm onSubmit={this.handleRequestSubmit}/>
+            {/*<RequestPasswordResetForm onSubmit={this.handleRequestSubmit}/>*/}
```
Comment (kedarv) [inline src/pages/PasswordReset/index.js]: do we need this still?

File: src/reducers/reducer_user.js [hunk 9]
```diff
@@ -1,56 +1,58 @@
...
+        isHacker,
+        isExec,
+        token: action.token
+      };
+    case LOGOUT_USER:
+      return INITIAL_STATE;
+    case REQUEST_ME:
+      return { ...state, loading: true };
+    case RECEIVE_ME:
+      //todo: error checking
```
Comment (kedarv) [inline src/reducers/reducer_user.js:42]: is this still a todo item?

File: src/reducers/reducer_user.js [hunk 10]
```diff
@@ -1,56 +1,58 @@
...
-            return state;
-    }
-}
\ No newline at end of file
+export default function(state = INITIAL_STATE, action) {
+  switch (action.type) {
+    case LOGIN_FROM_JWT_SUCCESS:
+      let token_data = jwtDecode(action.token);
+      console.log(token_data);
+      console.log(action.token);
```
Comment (kedarv) [inline src/reducers/reducer_user.js]: remove logs

File: src/pages/Login/index.js [hunk 11]
```diff
@@ -1,74 +1,76 @@
...
+    d.append('email', email);
+    d.append('password', password);
+    return apiFetch('users/login', {
+      method: 'POST',
+      body: d
+    })
+      .then(response => response.json())
+      .then(json => {
+        if (json.success === false) {
+          throw new SubmissionError({ _error: json.message });
```
Comment (kedarv) [inline src/pages/Login/index.js:26]: I get a `Unhandled Rejection (SubmissionError): Submit Validation Failed` error when hitting this statement.

File: src/pages/Register/RegisterForm.js [hunk 12]
```diff
@@ -1,35 +1,81 @@
...
+          type="password"
+          className="topSpacing"
+          component={renderField}
+          label="Password"
+          value={this.state.password}
+          onChange={this.updatePassword}
+        />
+        {error && <div className="margint">{error}</div>}
+        <Button
+          full={true}
```
Comment (kedarv) [inline src/pages/Register/RegisterForm.js:61]: Fix ` Unknown prop `full` on <button> tag` error

**Review by kedarv — CHANGES_REQUESTED** [review]
Looks pretty good! Needs a couple console.log and unneeded code cleanups, and also a second look at some potential warnings/errors

File: src/components/Nav/Nav.js (hunk 4 above)
Comment (colejoh) [reply src/components/Nav/Nav.js:40]: There's no way to log out currently. Was using it to test logging in and out

File: src/pages/Login/LoginForm.js (hunk 5 above)
Comment (colejoh) [reply src/pages/Login/LoginForm.js:58]: Oh I had `&nbsp;` but I guess prettify did that

File: src/reducers/reducer_user.js (hunk 9 above)
Comment (colejoh) [reply src/reducers/reducer_user.js:42]: Looks like it

File: src/pages/Register/RegisterForm.js (hunk 12 above)
Comment (colejoh) [reply src/pages/Register/RegisterForm.js:61]: I don't actually know how to fix this.

File: src/pages/Login/LoginForm.js (hunk 5 above)
Comment (nickysemenza) [reply src/pages/Login/LoginForm.js:58]: yep ^

File: src/pages/PasswordReset/index.js (hunk 8 above)
Comment (nickysemenza) [reply src/pages/PasswordReset/index.js]: we got a lot of emails until i added this

File: src/reducers/reducer_user.js (hunk 9 above)
Comment (nickysemenza) [reply src/reducers/reducer_user.js:42]: 😄

File: src/pages/Login/index.js (hunk 11 above)
Comment (colejoh) [reply src/pages/Login/index.js:26]: I cannot reproduce

File: src/pages/Login/LoginForm.js (hunk 5 above)
Comment (kedarv) [reply src/pages/Login/LoginForm.js:58]: ++

**Pull request merged**